tests/e2e/spark-versions/chainsaw-test.yaml

```diff
@@ -0,0 +1,98 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+  name: spark-operator-spark-version-validation
+spec:
+  scenarios:
+  - bindings:
+      - name: "SPARK_VERSION"
+        value: "3.5.2"
+      - name: "SCALA_VERSION"
+        value: "2.12"
+      - name: "JAVA_VERSION"
+        value: "17"
+      - name: "IMAGE"
+        value: 'spark:3.5.2-scala2.12-java17-python3-r-ubuntu'
+  - bindings:
+      - name: "SPARK_VERSION"
+        value: "3.4.3"
+      - name: "SCALA_VERSION"
+        value: "2.12"
+      - name: "JAVA_VERSION"
+        value: "11"
+      - name: "IMAGE"
+        value: 'spark:3.4.3-scala2.12-java11-python3-r-ubuntu'
+  - bindings:
+      - name: "SPARK_VERSION"
+        value: "4.0.0-preview1"
```

Comment:
The ordering seems to be a little weird because it's not sorted: `3.5.2`, `3.4.3`, `4.0.0-preview1`.

I'd recommend to put the higher version first: `4.0.0-preview`, `3.5.2`, `3.4.3`.